cmd/kops/create_cluster.go

```diff
@@ -727,6 +727,17 @@ func RunCreateCluster(ctx context.Context, f *util.Factory, out io.Writer, c *Cr
 				// Don't wrap file-not-found
 				if os.IsNotExist(err) {
 					klog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
+					// Try to look for ed25519 type key
+					sshPublicKeyPath = "~/.ssh/id_ed25519.pub"
+					c.SSHPublicKeys, err = loadSSHPublicKeys(sshPublicKeyPath)
+					if err != nil {
+						// Don't wrap file-not-found
+						if os.IsNotExist(err) {
+							klog.V(2).Infof("ssh key not found at %s", sshPublicKeyPath)
+						} else {
+							return fmt.Errorf("error reading SSH key file %q: %v", sshPublicKeyPath, err)
+						}
+					}
```

Comment:
This should be a separate PR.
Please put both keys in an array and loop over it. Stop at the first found.

Reply:
I opened a new PR #14638 for theses lines alone.